### PR TITLE
std.csv assert messages

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

**Comment by burner** [comment]
@thewilsonator thank you

File: std/csv.d [hunk 1]
```diff
@@ -989,7 +989,7 @@ public:
      */
     @property auto front()
     {
-        assert(!empty);
+        assert(!empty, "An empty range has no front element");
```
Comment (jondegenhardt) [inline std/csv.d]: More consistent with the messages in range code (std/range/package.d) would be:
```
"Attempting to fetch the front of an empty CsvReader"
```

File: std/csv.d [hunk 2]
```diff
@@ -1224,7 +1224,7 @@ public:
      */
     @property Contents front() @safe pure
     {
-        assert(!empty);
+        assert(!empty, "An empty range has no front element");
```
Comment (jondegenhardt) [inline std/csv.d]: More consistent with the messages in range code (std/range/package.d) would be:
```
"Attempting to fetch the front of an empty CsvRecord"
```

File: std/csv.d [hunk 3]
```diff
@@ -1425,7 +1425,7 @@ if (isSomeChar!Separator && isInputRange!Range
 
     while (!input.empty)
     {
-        assert(!(quoted && escQuote));
+        assert(!(quoted && escQuote), "Invalid quotation state");
```
Comment (jondegenhardt) [inline std/csv.d]: Consider adding the function to the message in some fashion. This is often helpful when debugging. eg.
```
"Invalid quotation state in csvNextToken"
```

Comment (jondegenhardt) [reply std/csv.d]: Just realized this is part of a much larger effort to add messages to all asserts. (Fantastic!) So, ignore my comment about this specific case and instead be consistent across all cases.

FWIW, I've benefitted when the assert messages include a reference to the code construct that is asserting. This has even been helpful to tracking down bugs in Phobos. But most important is that there is an assert message.

**Comment by burner** [comment]
@jondegenhardt thank you for the review

**Comment by jondegenhardt** [comment]
LGTM. Thanks for making the changes.

**Comment by burner** [comment]
@jondegenhardt that's what review is for

**Comment by thewilsonator** [comment]
@burner please do remind me about these when I forget, (unless you've forgotten too...).

**Comment by burner** [comment]
@thewilsonator sure thing, thanks

**Pull request merged**